system/nagios-status-check/detectors-nagios-status-check.tf

```
@@ -0,0 +1,27 @@
+resource "signalfx_detector" "status_check" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Custom script status"
+
+  program_text = <<-EOF
+        signal = data('gauge.status', filter=filter('plugin', 'nagios') and ${module.filter-tags.filter_custom})${var.status_check_aggregation_function}.${var.status_check_transformation_function}(over='${var.status_check_transformation_window}').publish('signal')
+        detect(when(signal >= 1) and when(signal < 2)).publish('WARN')
+        detect(when(signal >= 2) and when(signal < 3)).publish('CRIT')
```

```suggestion
        detect(when(signal >= threshold(1)) and when(signal < 2)).publish('WARN')
        detect(when(signal >= threshold(2)) and when(signal < 3)).publish('CRIT')
```

---

And for UNKNOWN ?

Maybe you should write:
```suggestion
        detect(when(signal >= threshold(1)) and when(signal < 2)).publish('WARN')
        detect(when(signal >= threshold(2))).publish('CRIT')
```
Or create un UNKNOWN dedicated rule.

---

I don't know if it's worse having an UNKNOWN state as we do not have it in any other detectors. I was also thinking to not put an upper threshold to the CRITICAL state as if the script is really not working we can consider that we must be alerted.